src/H5.c

```diff
@@ -145,8 +136,13 @@ H5__init_package(void)
 herr_t
 H5_init_library(void)
 {
+    size_t i;
     herr_t ret_value = SUCCEED;
 
+    /* Run the library initialization routine, if it hasn't already run */
+    if (H5_INIT_GLOBAL || H5_TERM_GLOBAL)
+        HGOTO_DONE(SUCCEED)
```

Comment:
does that mean that if H5_term_global is set there's no need to initialize anymore ? it looks like once you finalized you can't initialize again? or it's no longer needed ?

Reply:
H5_TERM_GLOBAL is set before library termination and reset after, so the library can reinitialize.

The check in H5_init_library ensures that the library cannot start to reinitialize while it is terminating. In the New Initialization Order, there are fewer ways to enter initialization, so we may be able to eliminate the check sometime in the future.